### PR TITLE
[BugFix] Fix bdb inconsistent bug

### LINKED ISSUE #6404: The master ip is still its own ip when the master is restared
### Steps to reproduce the behavior (Required)
Execute the following steps in a 3 follower cluster without any tasks
1. Hang the two non-master nodes in the cluster
```
kill -19 pid
```

2. Immediately execute create database operation on the master node
```
create database test;
```

3. Wait for the master node to exit
4. Restart the hung fe
5. After the hung fe started, start the old master node.
6. connect to the old master node, and execute the command:
```
show frontends;
```

### Expected behavior (Required)
`show frontends` should be executed successfully.

### Real behavior (Required)
`show frontends` is hung, 
```
2022-05-20 14:53:25,691 INFO (thrift-server-pool-2978|3102) [MasterOpExecutor.forward():134] Forward statement 2980 to Master TNetworkAddress(hostname:172.26.92.155, port:29927)
2022-05-20 14:53:25,692 INFO (thrift-server-pool-2979|3103) [ThriftServerEventProcessor.createContext():98] create thrift context. client: TNetworkAddress(hostname:172.26.92.155, port:54664)
2022-05-20 14:53:25,692 INFO (thrift-server-pool-2979|3103) [FrontendServiceImpl.forward():649] receive forwarded stmt 2980 from FE: 172.26.92.155
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6404

## Problem Summary(Required) ：

In bdb, synchronizing data with the master node is an asynchronous process. If a node finds dirty data in its own data, the dirty data should be truncated, but the dirty data may have been read by our replayer thread. So we should catch the RollbackException and exit the process. When truncating the dirty data, if the data precedes a checkpoint that deleted log files, an InsufficientLogException will be thrown, so we also should catch the InsufficientLogException and exit the process.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 16 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/journal/bdbje/BDBEnvironment.java|0|11|00.00%|
|:large_blue_circle:|com/starrocks/journal/bdbje/BDBJEJournal.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBEnvironment.java [hunk 1]
```diff
@@ -230,21 +230,19 @@ public void setup() {
...
+        try {
+            NetworkRestore restore = new NetworkRestore();
+            NetworkRestoreConfig config = new NetworkRestoreConfig();
+            config.setRetainLogFiles(false); // delete obsolete log files.
+            // Use the members returned by insufficientLogEx.getLogProviders()
+            // to select the desired subset of members and pass the resulting
+            // list as the argument to config.setLogProviders(), if the
+            // default selection of providers is not suitable.
+            restore.execute(insufficientLogEx, config);
+        } catch (Throwable t) {
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBEnvironment.java:243]: if this catch Throwable will RollbackException not throw?

**Review by Astralidea — APPROVED** [review]

**Review by padmejin — APPROVED** [review]
I still think that these corner cases should be covered in our unittest, but this time I will approve it.

**Review by padmejin — APPROVED** [review]
I still think that these corner cases should be covered in our unittest, but this time I will approve it.

**Review by decster — APPROVED** [review]

**Pull request merged**